clients/rest-client-generator/src/main/java/uk/gov/justice/services/clients/rest/generator/strategy/ClientGenerationStrategyFactory.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.services.clients.rest.generator.strategy;
+
+import static java.lang.Boolean.parseBoolean;
+
+import uk.gov.justice.raml.core.GeneratorConfig;
+
+import java.util.Map;
+
+public final class ClientGenerationStrategyFactory {
+
+    private ClientGenerationStrategyFactory() {
+    }
+
+    /**
+     * Factory for creating a {@link ClientGenerationStrategy} from the GeneratorConfig properties
+     * Returns either a {@link ClientActionMappingGeneration} or {@link ClientMediaTypeGeneration}
+     * strategy. This is set using the actionMapping generator property.
+     *
+     * actionMapping set to true returns - {@link ClientActionMappingGeneration}
+     *
+     * actionMapping set to false or not set returns -  {@link ClientMediaTypeGeneration}
+     *
+     * @param generatorConfig the generator configuration to check
+     * @return the selected ClientGenerationStrategy
+     */
+    public static ClientGenerationStrategy createFrom(final GeneratorConfig generatorConfig) {
+        if (isActionMapping(generatorConfig)) {
+            return new ClientActionMappingGeneration();
+        }
+        return new ClientMediaTypeGeneration();
+    }
+
+    private static boolean isActionMapping(final GeneratorConfig configuration) {
+        final Map<String, String> generatorProperties = configuration.getGeneratorProperties();
```

Comment:
do we need to pull this out into a separate variable, no null check etc.

Is this duplicated code from similar comment above or just 'similar' thing we are doing? can't remember exactly it was about 2000 lines ago :D